src/main/java/build/buildfarm/worker/Executor.java

```diff
@@ -112,18 +124,26 @@ public void run() {
                 .setResult(resultBuilder.build())
                 .build()))
             .build();
-        owner.output().offer(new OperationContext(
+        owner.output().put(new OperationContext(
             operation,
             operationContext.execDir,
             operationContext.metadata,
             operationContext.action,
             operationContext.inputFiles,
             operationContext.inputDirectories));
       } else {
-        owner.error().offer(operationContext);
+        try {
+          owner.error().put(operationContext);
+        } catch (InterruptedException intEx) {
+          Thread.currentThread().interrupt();
+        }
       }
     } catch (InterruptedException ex) {
```

Comment:
I think another Thread.currentThread().interrupt(); is missing here. In fact, I think it should replace the owner.error().put(operationContext); If you're getting an interrupt, you don't want to start another potentially long operation as a response to the interrupt, you just want to exit. Ditto with line 107 -- you might want to split them.

Reply:
Agreed. I'm splitting this into a single runInterruptible() that can throw InterruptedException and bail